### PR TITLE
Multiple Importmaps

### PR DESCRIPTION
**Pull request by henrikbjorn**
Recently the `preload` option was merged. But it comes with a huge penalty for a mental model. It is required for each dependency to set the preload for entrypoints. Depending on the number of items it can be hard to follow.

Instead I believe this complexity can be solved with allowing multiple importmaps. But by default just work with a single one like today.

This removes the mental headache since a the dependencies in `config/importmaps/admin.rb` can be seen as a group. Without reading each `preload` option.

This also allows to silo each importmap and reduce different dependencies to interfere with each other (e.g admin vs application).

### CODE REVIEW

**Comment by henrikbjorn** [comment]
@dhh would you consider this implementation, since it won't change anything regarding the current usage?

**Comment by henrikbjorn** [comment]
This could also make 3rd party gems such as ActiveAdmin make it easier to work with their importmaps. As they can create a `config/importmaps/active_admin.rb` as part of the install and then allow developers to work with it through the CLI `bin/importmap -i active_admin pin json_view`

**Comment by vfonic** [comment]
We are having this conversation in ActiveAdmin discussion thread here: https://github.com/activeadmin/activeadmin/discussions/8694

This would allow us to separate ActiveAdmin from frontend importmap. Currently, ActiveAdmin v4 requires importmap-rails in order to work. The issue is that `<%= javascript_importmap_tags %>` "leaks" the entire `config/importmap.rb` mapping wherever it's included. More info in the discussion thread.

How can we push this forward?
